### PR TITLE
UI: Fix shortcut button focus border to support high contrast

### LINKED ISSUE #13373: Shortcut button focus is not visible in high contrast mode
1. go to https://next--storybookjs.netlify.app/official-storybook/?path=/story/addons-a11y-basebutton--default
2. turn on high contrast mode in windows
3. turn of "focus" element state
4. note that there is no visible focus indicator

image

### PR DESCRIPTION
**Pull request by micahgodbolt**
Issue: #13373 

box shadows are not an accessible way to add focus to a button because shadows are dropped by high contrast mode. Using a pseudo element with border avoids the issue of borders messing with box sizing. 

image

### CODE REVIEW

**Comment by shilman** [comment]
Thanks @micahgodbolt! @tooppaaa @domyen @jsomsanith can I get a review?

**Comment by shilman** [comment]
Hi @adagar any chance I can get a review for this?

**Comment by adagar** [comment]
@shilman sure, I can take a look at this sometime today!

File: lib/ui/src/components/sidebar/Menu.tsx [hunk 1]
```diff
@@ -47,7 +47,17 @@ export const MenuButton = styled(Button)<MenuButtonProps>(({ highlighted, theme
   padding: 7,
```
Comment (domyen) [inline lib/ui/src/components/sidebar/Menu.tsx:47]:
```suggestion
  padding: 7,
  transition: none; //prevents button border from flashing when focused/blurred
```

File: lib/ui/src/components/sidebar/Menu.tsx [hunk 2]
```diff
@@ -47,7 +47,17 @@ export const MenuButton = styled(Button)<MenuButtonProps>(({ highlighted, theme
...
   '&:focus': {
     background: theme.barBg,
-    boxShadow: `${theme.color.secondary} 0 0 0 1px inset`,
+    boxShadow: 'none',
+  },
+  '&:focus:before': {
```
Comment (domyen) [inline lib/ui/src/components/sidebar/Menu.tsx:52]: This solution wont be immediately obvious for future maintainers. Can you add a comment that describes what's going on here that links to this PR?

**Review by domyen — CHANGES_REQUESTED** [review]
Thanks @micahgodbolt! Looks good, suggestions below.

File: lib/ui/src/components/sidebar/Menu.tsx (hunk 2 above)
Comment (adagar) [reply lib/ui/src/components/sidebar/Menu.tsx:53]: Agreed; this approach of creating a before/after selector to make an additional border is really useful, so documenting it out will help future devs a lot!

File: lib/ui/src/components/sidebar/Menu.tsx [hunk 3]
```diff
@@ -45,9 +45,20 @@ export const MenuButton = styled(Button)<MenuButtonProps>(({ highlighted, theme
   position: 'relative',
   overflow: 'visible',
   padding: 7,
+  transition: none; //prevents button border from flashing when focused/blurred
```
Comment (adagar) [inline lib/ui/src/components/sidebar/Menu.tsx:48]: This line is creating errors when I try to run it locally

```SyntaxError: storybook/lib/ui/src/components/sidebar/Menu.tsx: Unexpected token, expected "," (48:18)```

When I try to change it to a `,` it tells me that `none` is not defined.

**Comment by adagar** [comment]
Had an error preventing builds from working locally, but not sure if I'm doing something wrong. Additionally: How do you activate the high-contrast setting you show in the screenshot above? Is it the windows built-in approach?

File: lib/ui/src/components/sidebar/Menu.tsx (hunk 3 above)
Comment (shilman) [reply lib/ui/src/components/sidebar/Menu.tsx:48]: what if you change it to:

```
transition: 'none',
```

File: lib/ui/src/components/sidebar/Menu.tsx (hunk 2 above)
Comment (micahgodbolt) [reply lib/ui/src/components/sidebar/Menu.tsx:53]: We use this approach exclusively on Fluent UI controls. Might be good to open up an issue to standardize on this approach across the repo.

**Review by shilman — APPROVED** [review]
LGTM! Thanks @micahgodbolt @adagar @domyen !!!

**Pull request merged**